tests/test_mesh_methods.py

```diff
@@ -0,0 +1,227 @@
+                2,
+                2,
+                0,
+                0,
+            ],
+            [
+                0,
+                0,
+                1.333333,
+                0.666666,
+                2,
+                2,
+                0.666666,
+                1.333333,
+                0.666666,
+                1.333333,
+                0.666666,
+                1.333333,
+            ],
+        ]
+    )
+    "Translating the mesh"
+    mockmesh.translate(target_center)
+
+    "Testing the translation"
+    assert_allclose(mockmesh.mesh_center, target_center)
+    assert_allclose(mockmesh.bounds, target_bounds)
+    assert_allclose(mockmesh.face_centers, target_face_centers, atol=1e-6)
+
+
+def test_mesh_scale():
+    """
+    This function tests the scaling of the mesh.
+    """
+    mockmesh = cube_mesh_init()
+    scaling_factor = np.array([2, 2, 2])
+
+    """
+    This is pyvista's numpy array that stores the bounds of the mesh.
+    format: [xmin, xmax, ymin, ymax, zmin, zmax]
+    since we scaled the uniform cube with edge 2 by 2 situated with center at [0, 0, 0],
+        the new bounds are: [-2,2,-2,2,-2,2]
+    """
+    target_bounds = np.array([-2, 2, -2, 2, -2, 2])
+
+    """
+    Similarly the face centers will also be translated from initial position,
+    or multiplied by 2;
+    checkout initial position in test_mesh_init.py.
+    Since mesh face centers are calculated using mesh vertices,
+    if the face centers are correct, the mesh vertices are also correct.
+    """
+    target_face_centers = np.array(
+        [
+            [
+                -0.666666,
+                0.666666,
+                -2,
+                -2,
+                0.666666,
+                -0.666666,
+                2,
+                2,
+                0.666666,
+                -0.666666,
+                0.666666,
+                -0.666666,
+            ],
+            [
+                -0.666666,
+                0.666666,
+                -0.666666,
+                0.666666,
+                -0.666666,
+                0.666666,
+                -0.666666,
+                0.666666,
+                2,
+                2,
+                -2,
+                -2,
+            ],
+            [
+                -2,
+                -2,
+                0.666666,
+                -0.666666,
+                2,
+                2,
+                -0.666666,
+                0.666666,
+                -0.666666,
+                0.666666,
+                -0.666666,
+                0.666666,
+            ],
+        ]
+    )
+
+    "Scaling the mesh"
+    mockmesh.scale(scaling_factor)
+    assert_allclose(mockmesh.bounds, target_bounds)
+    assert_allclose(mockmesh.face_centers, target_face_centers, atol=1e-6)
+
+
+def test_mesh_rotate():
+    """
+    This function tests the rotation of the mesh.
+    """
+    mockmesh = cube_mesh_init()
+    rotation_angle = 90.0
+    rotation_axis = np.array([1, 0, 0])
```

Comment:
What happens when you give it a non-unit vector? does it just use the direction of the vector? or does it normalize and scale the rotation angle by the magnitude?

Reply:
No, it just uses the direction of the vector to form the rotation axis.